source/configuration_manual/mail_location/index.rst

```diff
@@ -1,299 +1,367 @@
+writable to the logged in user, or create the user's directory with proper
+permissions before the user logs in.
 
-.. code-block:: none
+Yu can disable the index files completely by appending ``:INDEX=MEMORY``. This
+is not recommended for production use, as the index files will need to be
+generated on every access.
 
-   # single-dbox
-   mail_location = sdbox:~/dbox
+Private Index Files
+^^^^^^^^^^^^^^^^^^^
 
-   # OR multi-dbox
-   mail_location = mdbox:~/mdbox
+.. versionadded:: v2.2
 
-Use only absolute paths. Even if relative paths would appear to work, they might just as well break some day.
+The recommended way to enable private flags for shared mailboxes is to create
+private indexes with ``:INDEXPVT=<path>``. See :ref:`public_shared_mailboxes`
+for more information.
 
-Directory hashing
-^^^^^^^^^^^^^^^^^^^
-You can use two different kinds of hashes in :ref:`config_variables` :
+INBOX Path
+----------
 
-* %H modifiers returns a 32bit hash of the given string as hex. 
+INBOX path can be specified to exist elsewhere than the rest of the mailboxes.
 
 Example:
 
 .. code-block:: none
 
-   %2.256H would return max. 256 different hashes in range 00 .. ff.
+   mail_location = maildir:~/Maildir:INBOX=~/Maildir/.INBOX
 
-* %M returns a MD5 hash of the string as hex. This can be used for two level hashing by getting substrings of the MD5 hash. 
+Autodetection
+=============
 
-Example:
+By default the :ref:`setting-mail_location` setting is empty, which means that
+Dovecot attempts to locate automatically where your mails are. This is done by
+looking, in order, at:
 
-.. code-block:: none
+* ~/mdbox/
+* ~/sdbox/
+* ~/Maildir/
+* ~/mail/.imap/
+* ~/mail/inbox
+* ~/mail/mbox
+* ~/Mail/.imap/
+* ~/Mail/inbox
+* ~/Mail/mbox
 
-   %1Mu/%2.1Mu/%u returns directories from 0/0/user to f/f/user.
+For autodetection to work, one of the above locations has to be populated;
+when autodetection is active, Dovecot will not attempt to create a mail folder.
 
-.. _Mail Location Index Files:
+.. Note::
 
-Index files
-^^^^^^^^^^^
-Index files are by default stored under the same directory as mails. With maildir they are stored in the actual maildirs, with mbox they are stored under ``.imap/`` directory. You may want to change the index file location if you're using :ref:`nfs` or if you're setting up :ref:`shared_mailboxes`.
+   .imap is a directory, and inbox and mbox are files.
 
-You can change the index file location by adding ``:INDEX=<path>`` to mail_location. For example:
+It's usually a good idea to explicitly specify where the mails are, even if
+the autodetection happens to work, in particular to benefit from auto-creation
+of the folder for new users.
 
-.. code-block:: none
+Custom Autodetection
+--------------------
 
-   mail_location = maildir:~/Maildir:INDEX=/var/indexes/%u
+If you need something besides the default autodetection, you can override the
+executable program to run a script, which sets the MAIL environment properly.
```

Comment:
No more executable program overwriting. Need to use post-login scripting (<post_login_scripting>)